engines/suppliers/app/controllers/suppliers/api/v1/producers_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Suppliers
+  module Api
+    module V1
+      class ProducersController < ApplicationController
+        before_action :authenticate
+        before_action :load_group,
+          only: [:index],
+          if: -> { !!params.fetch('group_id', nil) }
+        before_action :load_producer, only: [:show]
+
+        # GET /api/v1/producers
+        #
+        # Without parameters it will return all the producers which the
+        # user is associated to directly as a member.
+        #
+        # Passing `group_id` parameter will return:
+        #  - the producers associated to the group through `Membership`
+        #  - the producers associated to the group through `Supplier`
+        #
+        def index
+          producers = ProducersCollection.new(user: current_user, group: @group).build
```

Comment:
Not sure we can do all in one query since we touch 3 tables: `producers`, `suppliers` and `memberships`. But we can investigate... just not now :sweat_smile: